service/s3/endpoints.go

```diff
@@ -132,27 +132,20 @@ func removeResolveEndpointMiddleware(stack *middleware.Stack) error {
 
 type wrappedEndpointResolver struct {
 	awsResolver aws.EndpointResolverWithOptions
-	resolver    EndpointResolver
 }
 
 func (w *wrappedEndpointResolver) ResolveEndpoint(region string, options EndpointResolverOptions) (endpoint aws.Endpoint, err error) {
-	if w.awsResolver == nil {
-		goto fallback
-	}
 	endpoint, err = w.awsResolver.ResolveEndpoint(ServiceID, region, options)
 	if err == nil {
 		return endpoint, nil
 	}
 
-	if nf := (&aws.EndpointNotFoundError{}); !errors.As(err, &nf) {
-		return endpoint, err
+	if nf := (&aws.EndpointNotFoundError{}); errors.As(err, &nf) {
+		return endpoint, nil
```

Comment:
question: Does downstream code handle this properly? i.e. if the v1 middleware gets a zero-value endpoint, it does nothing and lets v2 pick up the rest?

Reply:
it didnt. pushed new commits with a fix. and tested via debugger with sample program and unit tests. heres my sample program:
```go
package main

import (
	"context"
	"log"
	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/service/s3"
)

func main() {
	customResolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
		return aws.Endpoint{}, &aws.EndpointNotFoundError{}
	})
	
	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithEndpointResolverWithOptions(customResolver), config.WithRegion("us-west-2"))

	// Create an Amazon S3 service client
	client := s3.NewFromConfig(cfg)

	// Get the first page of results for ListObjectsV2 for a bucket
	_, err = client.GetObject(context.TODO(), &s3.GetObjectInput{
		Bucket: aws.String("test-bucket"),
		Key: aws.String("test-key"),
	})
	if err != nil {
		log.Fatal(err)
	}

}
```